compiler/noirc_driver/src/contract.rs

```diff
@@ -51,4 +51,7 @@ pub struct ContractFunction {
     pub bytecode: Program,
 
     pub debug: DebugInfo,
+
+    /// Names of the functions in the program. These are used for more informative debugging and benchmarking.
+    pub names: Vec<String>,
```

Comment:
Could these names be defined with the functions themselves, as in `SsaCircuitArtifact`, so that they're always 1-to-1?